odra-schema/src/ty.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::BTreeMap;
+
+pub use casper_contract_schema;
+use casper_contract_schema::NamedCLType;
+use casper_types::{bytesrepr::Bytes, Key, PublicKey, URef, U128, U256, U512};
+use odra_core::{args::Maybe, Address};
+
+pub trait NamedCLTyped {
```

Comment:
Shouldn't this be in `casper_contract_schema`?

Reply:
I would say no. We need to implement this for all `CLType`s and `Maybe`.  `casper_contract_schema` does not know about `Maybe` (and shoudn't know) and you can't implement an external trait for an external type.

Reply:
If `odra_core` depends on `casper_contract_schema`, then you can implement `NamedCLType` for `Maybe` next to `Maybe`'s definition.

That being said, after thinking about it, we can leave it like you implemented and come back to it, when someone else actually use `casper_contract_schema` crate.

Reply:
But `odra-core` does not depend on `casper_contract_schema` only `odra-schema` does. But we come back to it if needed.